components/TheHeader.vue

```diff
@@ -46,25 +46,42 @@
             <!-- Main nav menu -->
             <nav v-if="menu" class="w-full lg:w-auto hidden lg:flex">
               <ul class="flex justify-center">
-                <li v-for="item in menu.items" :key="item.name" class="sw-nav-link-wrapper mb-0">
+                <li
+                  v-for="(item, index) in menu.items"
+                  :key="item.name"
+                  class="sw-nav-link-wrapper mb-0"
+                >
                   <NuxtLink
                     :to="resolveUrl(item)"
                     :title="item.description"
                     class="sw-nav-link relative flex items-center h-full px-5 pt-1 rounded-none border-transparent border-b-4"
                     @click.native="megaNavIsEnabled = false"
-                    @mouseleave.native="resetMegaNav"
+                    @mouseleave.native="hideMegaNav"
+                    @mouseenter.native="showMegaNav(index)"
                   >
                     <span class="relative">
                       {{ item.name }}
                     </span>
                   </NuxtLink>
                   <!-- Show mega nav if item has child items -->
-                  <div
-                    v-if="item && item.items && megaNavIsEnabled"
-                    class="mega-nav fade-in hidden absolute left-0 right-0 min-h-full"
-                  >
-                    <MegaNav :items="item.items" @click="resetMegaNav" />
-                  </div>
+                  <transition name="fade">
+                    <div
+                      v-if="
+                        (item &&
+                          item.items &&
+                          item.items.length &&
+                          megaNavIsEnabled &&
+                          currentMegaNavIndex === index) ||
+                          !isMounted
+                      "
```

Comment:
@by-tim can you extract this conditional into a method and include some comments on what is being checked?

Reply:
Done with latest commit.